File: lib/src/normalize_employee_name.dart
String normalizeEmployeeName(String input) {
  input = input.trim();
  if (input.isNotEmpty) {
    String result = input
        .split(" ")
        .where((e) => e != "")
        .map((e) => e[0].toUpperCase() + e.substring(1).toLowerCase())
        .join(" ");
    return result;
  }
  return "";
}

File: javachallenges/src/main/java/challenges/NormalizeEmployeeName.java
package challenges;

import java.util.Arrays;

import java.util.stream.Collectors;

public class NormalizeEmployeeName {
    public static String NormalizeEmployeeName(String input){

        input=input.trim();
        if(!input.isEmpty()){
            return Arrays.stream(input.split("\\s+")).map((x)->(x.substring(0,1).toUpperCase()+x.substring(1).toLowerCase())).collect(Collectors.joining(" "));
        }
        else{
            return "invalid input";
        }

    }
}
